calendar_resource/models/calendar_event.py

```diff
@@ -178,15 +169,33 @@ def _check_resource_ids_leaves(self):
                     _(
                         "The resource, '%s', is on leave during "
                         "the following times which are conflicting with "
-                        "this event.\n%s"
+                        "this event.\n%s",
                     )
                     % (
                         resource.name,
-                        self._format_datetime_interval_list(datetimes),
+                        self._format_datetime_intervals_to_str(datetimes),
                     )
                 )
 
     @api.multi
+    def _get_event_date_list(self):
+        self.ensure_one()
+        start = fields.Date.from_string(self.start)
+        stop = fields.Datetime.from_string(self.stop)
+
+        if stop.time() == time(0, 0):
+            stop -= timedelta(days=1)
+
+        stop = stop.date()
+        date = start
+        dates = []
+        while date <= stop:
```

Comment:
IMO it's best to avoid `while` and `until` loops whenever possible - they can cause infinite loops because there is no explicit ending. The main time I make an exception is something like:

```
arr = [1, 2, 3]
while len(arr):
  val = arr.pop()
  if val % 2:
    arr.append(val * 2)
```

In the above, we technically have an explicit ending because the list is sure to end at some point. We also couldn't use the standard range iteration tactic, because the list is being mutated from underneath the loop. Neither of those conditions apply here.

Maybe use a [`dateutil.rrule`](https://dateutil.readthedocs.io/en/stable/rrule.html) instead?